lib/bolt/cli.rb

```diff
@@ -587,8 +591,27 @@ def list_plans
     end
 
     def list_targets
+      inventoryfile = config.inventoryfile || config.default_inventoryfile
+
+      # Check target args against group and target names in the inventory. This
+      # needs to be done separately from resolving the inventory as update_targets
+      # adds targets to the inventory.
+      targets       = resolve_target_args(options).first&.split(',') || []
```

Comment:
Can we just build the `known_names` set before calling `update_targets`? I don't quite follow why we _have_ to do this step separately.

Reply:
Err, definitely. Updated. :)